fix: Add Underline Font Style and Label Style

jdun28 opened the pull request:
> Pull Request Template
>
> Use this template to quickly create a well written pull request. Delete all quotes before creating the pull request.

## Description

Adds the icon and value for fontstyles and labelstyles in various widgets.

Fixes # (issue)

Fixes: #9403 

## Type of change

> Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Updated `app/client/src/components/propertyControls/ButtonTabControl.test.tsx` to include passing in underline as a option that is passed in.


## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

jdun28 commented:
@aswathkk I'm not sure what to do about the failures on this PR but I believe I have it all ready to go. I added to a few extra places that were not outlined in the GH issue but where label styles or font styles were defined. If you want me to remove those extra ones, just let me know - it's no problem.

aswathkk commented on app/client/src/widgets/TableWidget/widget/propertyConfig.ts:
```diff
@@ -1705,6 +1705,10 @@ export default [
             icon: "ITALICS_FONT",
             value: "ITALIC",
           },
+          {
```
This is a V1 of the Table widget that is deprecated now. We don't want to add this property here.

aswathkk commented on app/client/src/widgets/MultiSelectWidget/widget/index.tsx:
```diff
@@ -335,6 +335,10 @@ class MultiSelectWidget extends BaseWidget<
                 icon: "ITALICS_FONT",
                 value: "ITALIC",
               },
+              {
```
MultiSelectWidget is deprecated we have a V2 for this widget. So, no need to add underline here!

aswathkk reviewed (changes requested):
Hi @jdun28 
Thanks for taking the time to contribute.

I found that you've added UNDERLINE to some of the deprecated widgets too. We don't want to do that since they are deprecated. I commented on a few places. There are some more widgets that are deprecated and I see that you've modified them too. Please take a look at those.

Also, when I select underline for some widgets like Input Widget, RichTextEditor, Checkbox etc, I found that the label is not getting underlined. I think you've just added property but didn't modify the underlying component to accept underline. Please take a look at this too

jdun28 commented:
@aswathkk I removed the underlines I added on the widgets I saw that also had a V2. If there are others that are deprecated, I'd need those listed out as I wasn't seeing any other indication of deprecation in the code.

Anywhere in the other widgets where the component contained something similar to
```
font-size: ${labelTextSize ?? "inherit"};
  font-weight: ${labelStyle?.includes(FontStyleTypes.BOLD) ? "bold" : "normal"};
  font-style: ${
    labelStyle?.includes(FontStyleTypes.ITALIC) ? "italic" : "normal"
  };
```
I added the text-decoration piece for `FontStyleTypes.UNDERLINE` however not every component had a handler like this and based off of a review of the components it looked like some only needed the style added to the Emphasis. If there are still ones that aren't working properly but their Bold/Italic options work please point me at those specifically so I can dig in further.

Thanks!

aswathkk commented:
@jdun28 
The list of Deprecated widgets is available here: [src/utils/WidgetRegistry.tsx](https://github.com/appsmithorg/appsmith/blob/c3e807d246812e3f188e790722a78af4cd1efe60/app/client/src/utils/WidgetRegistry.tsx#L217-L225)

Whereever you added underline option, please verify yourself whether the option is working as expected or not. I found most of them aren't working.
